src/util.rs

```diff
@@ -33,3 +35,21 @@ pub fn bytes_diff<'a, 'b>(base: &'a [u8], target: &'b [u8]) -> &'b [u8] {
         target.get_unchecked(end..)
     }
 }
+
+/// simple rewrite of golang sort.Search
+pub fn search<F>(n: usize, mut f: F) -> usize
```

Comment:
can we use `slice.binary_search_by` directly?

Reply:
The goal of using this customized search function is that,

* We need to search on a number range `0..n`, instead of on a slice of concrete data. Data will be loaded into memory upon each search iteration.
* It seems that we could not directly use `binary_search_by` on a `Range` object.

Reply:
FYI, https://github.com/tikv/agatedb/blob/462e28dfb5445d1f2a97a7cfee02cd51c88dc87a/src/table/iterator.rs#L133